cmake/kokkos_arch.cmake

```diff
@@ -1105,6 +1105,8 @@ if(KOKKOS_ENABLE_SYCL)
     elseif(KOKKOS_ARCH_INTEL_DG1)
       set(SYCL_TARGET_BACKEND_FLAG -Xsycl-target-backend "-device dg1")
     elseif(KOKKOS_ARCH_INTEL_DG2)
+      unset(SYCL_TARGET_FLAG)
+      set(SYCL_TARGET_FLAG -fsycl-targets=spir64)
```

Comment:
This ultimately means that ahead-of-time compilation doesn't work but just-in-time-compilation does.
In any case, this looks like the wrong fix. Either `device` flag for the GPU you are testing with is wrong (but it seemed to work fine for me cross-compiling on `Aurora`) or AOT compilation just doesn't work for whatever reason and you shoud use `Kokkos_ARCH_INTEL_GEN=ON` instead of `Kokkos_ARCH_INTEL_DG2=ON`.

Reply:
AOT crashes the compiler. The Intel team suggested using runtime compilation on the sycl branch. I added the device flag and didn't observe issues on the risc-v target.

Reply:
Right, `Kokkos_ARCH_INTEL_GEN=ON` is the option to use for JIT compilation. No need to change the AOT flags.

Reply:
So close out this PR?

Reply:
Yes, let's close. I assume the AOT issue is tracked on Intel's side.